synapse/storage/controllers/state.py

```diff
@@ -485,7 +485,7 @@ async def get_canonical_alias_for_room(self, room_id: str) -> Optional[str]:
         if not event:
             return None
 
-        return event.content.get("canonical_alias")
+        return event.content.get("alias")
```

Comment:
Looks like this has just always been wrong? The [spec](https://spec.matrix.org/v1.7/client-server-api/#mroomcanonical_alias) seems to say this is `alias`. I wonder what badness thias hs brought.

It looks like this is only used in [one spot](https://github.com/matrix-org/synapse/blob/2e59e97ebd02e93da39e6c90335d3b24ed01217a/synapse/handlers/directory.py#L488-L492), the impact of which would be that we would not properly check the canonical alias vs. the synapse configuration for if publishing of the room is allowed.

Reply:
This sounds exactly right, I changed it [after @DMRobertson mentioned it would be a good idea](https://github.com/matrix-org/synapse/pull/15450#discussion_r1169372579)